### PR TITLE
Add .gitpod.yml to gitignore

### PR DESCRIPTION
**Pull request by Pandapip1**
This allows for GitPod (https://gitpod.io/) to be used for editing, without polluting the repository with a `.gitpod.yml` file. It is currently not possible to disable this feature, so the alternative to adding it to the gitignore is manually deleting the file every other commit.

### CODE REVIEW

**Comment by Pandapip1** [comment]
@MicahZoltu courtesy ping

File: .gitignore
Comment (MicahZoltu) [inline .gitignore:7]:
```suggestion
# https://www.gitpod.io/docs/config-gitpod-file
```
Lets include a link to where people can get more information on this specific file type, rather than being generic/ambiguous.

**Review by MicahZoltu — APPROVED** [review]
Will wait for additional approvals from other editors, but I'm generally OK with this ignore.  Would like to see the comment changed before merging though (or an argument why we shouldn't).

cc @axic @lightclient @gcolvin @SamWilsn

**Comment by MicahZoltu** [comment]
Hmm, @alita-moore This shouldn't have auto-merged...

**Comment by MicahZoltu** [comment]
> file .gitignore is not a valid filename, but this error has been ignored due to editor approvals

This override should only happen if multiple editors approve.  In a perfect world, the number of editors required for approval would decrease over time.  If `n` editors approve it will merge immediately, if `n-1` editors approve then maybe it has to bake for a few days.  If `n-2` editors approve it perhaps has to bake for a week or two.  If `n-3` editors approve then it has to bake for a month, etc.

I don't know what `n` is.  4 perhaps?  If that is too complex, then we just either remove auto-merge of `ambiguous` changes entirely, or require something like 3 editor approvals before automerge.

**Comment by alita-moore** [comment]
> > file .gitignore is not a valid filename, but this error has been ignored due to editor approvals
> 
> This override should only happen if multiple editors approve.  In a perfect world, the number of editors required for approval would decrease over time.  If `n` editors approve it will merge immediately, if `n-1` editors approve then maybe it has to bake for a few days.  If `n-2` editors approve it perhaps has to bake for a week or two.  If `n-3` editors approve then it has to bake for a month, etc.
> 
> I don't know what `n` is.  4 perhaps?  If that is too complex, then we just either remove auto-merge of `ambiguous` changes entirely, or require something like 3 editor approvals before automerge.

Ah it's not super complicated but I can implement the requires 3 editors check much more easily

**Comment by Pandapip1** [comment]
Created a PR to revert this (#4870)

**Pull request merged**
